### PR TITLE
Quality of life improvements and better docs

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
### Features:
 - [x] Manual topic naming.
 - [x] Manual topic joining in clustering methods.
 - [x] $S^3$ can now be refit with a different number of components, iterations or seed way faster.
 - [x] Docs
 - [x] Serialization + HuggingFace Hub

### CODE REVIEW

File: turftopic/models/decomp.py [hunk 1]
```diff
@@ -130,6 +131,85 @@ def fit_transform(
...
         return doc_topic
 
+    def refit_transform(
+        self,
+        n_components: Optional[int] = None,
+        max_iter: Optional[int] = None,
+        random_state: Optional[int] = None,
+    ):
+        """Refits model with the given parameters.
+        This is significantly faster than fitting a new model from scratch.
```
Comment (KennethEnevoldsen) [inline turftopic/models/decomp.py:141]: Why

File: docs/GMM.md [hunk 2]
```diff
@@ -9,47 +9,51 @@ These Gaussian components are assumed to be the topics.
   <figcaption>Components of a Gaussian Mixture Model <br>(figure from scikit-learn documentation)</figcaption>
 </figure>
 
-## The Model
+## How does GMM work?
```
Comment (KennethEnevoldsen) [inline docs/GMM.md:12]: Specify shorthand

File: docs/persistence.md [hunk 3]
```diff
@@ -1,25 +1,27 @@
 # Model Persistence
 
-Similarly to scikit-learn we primarily recommend that you use the `joblib` package for efficiently serializing Turftopic models.
+Turftopic models can now be persisted to disk using the `to_disk()` method of models:
```
Comment (KennethEnevoldsen) [inline docs/persistence.md:3]:
```suggestion
Turftopic models can be saved to disk using the `to_disk()` method of models:
```

**Review by KennethEnevoldsen — APPROVED** [review]
Only a few minor edits

File: turftopic/models/decomp.py (hunk 1 above)
Comment (x-tabdeveloping) [reply turftopic/models/decomp.py:141]: Huh?

Comment (KennethEnevoldsen) [reply turftopic/models/decomp.py:141]: I would add documentation on why this is significantly faster
